tools/linkage_disequilibrium/ld_tool

```diff
@@ -0,0 +1,401 @@
+  'r2=f',                        # value between 0 and 1
+  'd_prime=f',                   # value between 0 and 1
+  'add_variant_attributes',      # add variant attributes: variant_evidence, variant_consequence  
+) or die "ERROR: Failed to parse command-line flags\n";
+
+&usage && exit(0) if (!$arg_count) || $config->{help};
+$config->{species} ||= $DEFAULT_SPECIES;
+$config->{assembly} ||= $DEFAULT_HUMAN_ASSEMBLY;
+&init_connections($config);
+&show_populations($config) && exit(0) if ($config->{show_populations});
+
+die "Select at least one calculation: ./ld_tool --calculation [region|center|pairwise]. Or use --help for more information." unless ($config->{calculation});
+
+&init_files($config);
+&init_populations($config);
+
+my $calculation = $config->{calculation};
+if ($calculation eq 'center') {
+  &center($config);
+} elsif ($calculation eq 'pairwise') {
+  &pairwise($config);
+} elsif ($calculation eq 'region') {
+  &region($config);
+} else {
+  die "Unsupported calculation. Use one of the following calculations: center, pairwise or region\n";
+}
+
+sub init_files {
+  my $config = shift;
+  my $input_file = $config->{input_file};
+
+  my $file_ending = {output_file => 'out', warnings_file => 'warnings'};
+
+  if (!$input_file) {
+    if (!@variants && !@regions) {
+      die "Input is missing. Either provide input data in a file with --input_file or use --variant [variant] or --region [region]. Use --help for more information.";
+    }
+    $input_file = &get_file_name($config);
+  }   
+  foreach my $file (qw/output_file warnings_file/) {
+    if ($config->{$file} && -e $config->{$file}) {
+      die "$file file already exists: ", $config->{$file};
+    } else {
+      $config->{$file} ||= $input_file . '.' . $file_ending->{$file};
+    }
+  }
+
+  if (@regions) {
+    if (!$config->{calculation} eq 'region') {
+      die "Region input must be used with --calculation region."; 
+    }
+  }
+
+  if (@variants) {
+    if (!($config->{calculation} eq 'center' || $config->{calculation} eq 'pairwise')) {
+      die "Variant input must be used with --calculation center or --calculation pairwise."; 
+    }
+  }
+
+}
+
+sub init_connections {
+  my $config = shift;
+  my $registry = 'Bio::EnsEMBL::Registry';
+  my $port = 3306;
+  my $assembly = lc $config->{assembly};
+  if ($assembly eq 'grch37') {
+    $port = 3337;
+  }
+  $registry->load_registry_from_db(
+    -host => 'ensembldb.ensembl.org',
+    -user => 'anonymous',
+    -port => $port,
+  );
+
+  my $species = $config->{species};
+  my $population_adaptor = $registry->get_adaptor($species, 'variation', 'population');
+  my $variation_adaptor = $registry->get_adaptor($species, 'variation', 'variation');
+  my $ld_feature_container_adaptor = $registry->get_adaptor($species, 'variation', 'ldfeaturecontainer');
+  $ld_feature_container_adaptor->db->use_vcf(1);
+  my $d_prime = $config->{'d_prime'};
+  if ($d_prime && ($d_prime < 0.0 || $d_prime > 1.0)) {
```

Comment:
I'd check the input options make sense right at the start before getting db connections.